packages/edit-site/src/components/dataviews/view-list.js

```diff
@@ -55,7 +57,7 @@ const sortingItemsInfo = {
 };
 const sortIcons = { asc: chevronUp, desc: chevronDown };
 
-function HeaderMenu( { dataView, header } ) {
+function HeaderMenu( { dataView, header, view, onChangeView } ) {
```

Comment:
What do you all think of using both TanStack and our own APIs in `HeaderMenu` for the time being?

Refactoring view list to use our own `view` instead of controlling Tanstack table's `state` via its own API is a big PR. We can piece it down by temporarily allowing both things to coexist here and removing bit by bit (column filters, sorting, pagination, visibility, search, etc.)

Reply:
If this is not a plan we can agree upon, I'll remove [67d5dec](https://github.com/WordPress/gutenberg/pull/56514/commits/67d5dec0eb24f8b30cda46a7ee631adf5e3eae96) from this PR and work on that separately,

Reply:
I'm ok with that personally, but let's add a task to follow-up on that and not leave it like that forever.